src/main/java/core/basesyntax/EmployeeService.java

```diff
@@ -1,10 +1,29 @@
 package core.basesyntax;
 
+import java.util.Comparator;
 import java.util.List;
 import java.util.Set;
+import java.util.TreeSet;
 
 public class EmployeeService {
     public Set<Employee> getEmployByOrder(List<Employee> employees) {
-        return null;
+        if (employees == null) {
+            throw new IllegalArgumentException("Input list cannot be null");
+        }
+        Comparator<Employee> employeeComparator = new Comparator<>() {
+            @Override
+            public int compare(Employee o1, Employee o2) {
+                int ageCompared = Integer.compare(o1.getAge(), o2.getAge());
+                if (ageCompared != 0) {
+                    return ageCompared;
+                }
+                return o1.getName().compareToIgnoreCase(o2.getName());
+            }
+        };
```

Comment:
The comparator is correctly implemented to first compare by age and then by name in a case-insensitive manner. Ensure that the `Employee` class implements the `Comparable` interface if you plan to use it in other sorted collections without a comparator.